codegen/src/api/metadata.rs

```diff
@@ -62,12 +62,8 @@ fn get_field_hash(
 ) -> [u8; 32] {
     let mut bytes = vec![MetadataHashableIDs::Field as u8];
 
-    if let Some(name) = field.name() {
-        bytes.extend(name.as_bytes());
-    }
-    if let Some(ty_name) = field.type_name() {
-        bytes.extend(ty_name.as_bytes());
-    }
+    bytes.extend(field.name().encode());
+    bytes.extend(field.type_name().encode());
```

Comment:
Ooh, actually a slightly more efficient approach for this I htink would be:

```
field.name().encode_by(&mut bytes);
field.type_name().encode_to(&mut bytes);
```

Just to avoid the intermediate vec allocations :)

Reply:
That makes a lot of sense! Thanks for the suggestion!